src/gt4py/next/program_processors/runners/dace/transformations/move_dataflow_into_if_body.py

```diff
@@ -398,6 +407,36 @@ def _replicate_dataflow_into_branch(
         inner_sdfg.arrays[connector].transient = True
         if_block.remove_in_connector(connector)
 
+    def _remove_outside_dataflow(
+        self,
+        sdfg: dace.SDFG,
+        state: dace.SDFGState,
+        relocatable_dataflow: dict[str, set[dace_nodes.Node]],
+    ) -> None:
+        """Removes the original dataflow, that has been relocated.
+
+        The function will also remove data containers that are no longer in use.
+        """
+        all_relocatable_dataflow: set[dace_nodes.Node] = functools.reduce(
+            lambda s1, s2: s1.union(s2), relocatable_dataflow.values(), set()
+        )
+
+        # Before we can clean the original nodes, we must clean the dataflow. If a
+        #  node, that was relocated, has incoming connections we must remove them
+        #  and the parent dataflow.
+        for node_to_remove in all_relocatable_dataflow:
+            for iedge in list(state.in_edges(node_to_remove)):
+                if iedge.src in all_relocatable_dataflow:
+                    continue
+                dace_sutils.remove_edge_and_dangling_path(state, iedge)
+
+            if isinstance(node_to_remove, dace_nodes.AccessNode):
+                assert node_to_remove.desc(sdfg).transient
+                sdfg.remove_data(node_to_remove.data, validate=False)
+
+        # Remove the original nodes.
```

Comment:
You first remove the array data, then the access node. That's fine, it works. Maybe just add to the line comment:
```suggestion
        # Remove the original nodes (the array data was already deleted in the above loop).
```